cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java

```diff
@@ -411,36 +430,40 @@ private JSONObject paramsToSendConsent(int actionType, Integer choiceId) throws
         }
     }
 
-    protected void sendConsent(int actionType, Integer choiceId) {
-        try {
-            sourcePoint.sendConsent(paramsToSendConsent(actionType, choiceId), new OnLoadComplete() {
-                @Override
-                public void onSuccess(Object result) {
-                    try {
-                        JSONObject jsonResult = new JSONObject((String) result);
-                        JSONObject jsonUserConsent = jsonResult.getJSONObject("userConsent");
-                        userConsent = new GDPRUserConsent(jsonUserConsent);
-                        euConsent = jsonUserConsent.getString("euconsent");
-                        consentUUID = jsonResult.getString("uuid");
-                        metaData = jsonResult.getString("meta");
-                        Log.i("GDPR_UUID", "From sendConsentReponse: " + consentUUID);
-                        consentFinished();
-                    } catch (Exception e) {
-                        Log.d(TAG, "Sorry, something went wrong");
-                        e.printStackTrace();
-                        onErrorTask(new ConsentLibException(e, "Error trying to parse response from sendConsents."));
+    protected void sendConsent(int actionType, Integer choiceId) throws  ConsentLibException {
+        if (!hasLostInternetConnection()) {
```

Comment:
we throw the exception directly instead of using a else clause

Reply:
yes will do this and follow same guidelines in future task.